foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_10_0/SparkPipelineMessagingMigration.java

```diff
@@ -0,0 +1,252 @@
+    private static final String AISSEMBLE_MESSAGING_CDI_IMPORT = "import com.boozallen.aissemble.messaging.core.cdi.MessagingCdiContext;";
+    private static final String AISSEMBLE_MESSAGING_CDI_OBJECT = "new MessagingCdiContext()";
+    private static final String AISSEMBLE_KAFKA_MESSAGING_CDI_IMPORT = "import com.boozallen.aissemble.kafka.context.KafkaConnectorCdiContext;";
+    private static final String AISSEMBLE_KAFKA_MESSAGING_CDI_OBJECT = "new KafkaConnectorCdiContext()";
+    private static final String AISSEMBLE_PIPELINE_CDI_OBJECT = "new PipelinesCdiContext()";
+ 
+    @Override
+    protected boolean shouldExecuteOnFile(File pomFile) {
+        // Check if the pipeline pom contains any smallrye-reactive-messaging dependencies
+        Model model = PomHelper.getLocationAnnotatedModel(pomFile);
+
+        if (this.hasDependency(model, SMALLRYE_REACTIVE_MESSAGING_GROUP_ID, SMALLRYE_REACTIVE_MESSAGING_ARTIFACT_ID) 
+            && model.getPackaging().equals("jar")) {
+
+            // Get the CdiContainerFactory.java file
+            File cdiContainerFactoryFile = searchForFileInCdi(new File(pomFile.getParentFile(), "src/main/java"), "CdiContainerFactory.java");
+
+            // Check if the pipeline pom contains the smallrye-reactive-messaging-kafka dependency
+            boolean addKafkaCdiContext = this.hasDependency(model, SMALLRYE_REACTIVE_MESSAGING_GROUP_ID, SMALLRYE_REACTIVE_MESSAGING_KAFKA_ARTIFACT_ID);
+
+            if (addKafkaCdiContext) {
+                // Check if any of the CdiContainerFactory.java or pom.xml needs to be migrated
+                return shouldMigratePomFile(model) || shouldMigrateCdiContainerFactoryFile(cdiContainerFactoryFile, addKafkaCdiContext);
+            }
+
+            // Check if only the CdiContainerFactory.java needs to be migrated
+            return shouldMigrateCdiContainerFactoryFile(cdiContainerFactoryFile, addKafkaCdiContext);
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    protected boolean performMigration(File pomFile) {
+        Model model = PomHelper.getLocationAnnotatedModel(pomFile);
+
+        // Get the CdiContainerFactory.java file
+        File cdiContainerFactoryFile = searchForFileInCdi(new File(pomFile.getParentFile(), "src/main/java"), "CdiContainerFactory.java");
+
+        // Check if the pipeline pom contains the smallrye-reactive-messaging-kafka dependency
+        boolean addKafkaCdiContext = this.hasDependency(model, SMALLRYE_REACTIVE_MESSAGING_GROUP_ID, SMALLRYE_REACTIVE_MESSAGING_KAFKA_ARTIFACT_ID);
+        
+        boolean cdiContainerFactoryFileMigrationSuccess = false;
+        if (shouldMigrateCdiContainerFactoryFile(cdiContainerFactoryFile, addKafkaCdiContext)) {
+            // Update the file with the new import(s) and CDI context object(s)
+            cdiContainerFactoryFileMigrationSuccess = this.migrateCdiContainerFactoryFile(cdiContainerFactoryFile, addKafkaCdiContext);
```

Comment:
S/I: We should avoid modifying files that aren't directly matched by Baton as much as possible. Baton creates back ups of files and can roll them back -- but that logic breaks when you violate the contract with Baton of only modifying files that your migration directly matched.

One way we could have done this would be to have a separate migration only for `CdiContainerFactory.java` that essentially does a check like:
```java
this.hasDependency(this.getMavenProject(), SMALLRYE_REACTIVE_MESSAGING_GROUP_ID, SMALLRYE_REACTIVE_MESSAGING_KAFKA_ARTIFACT_ID) && shouldMigrateCdiContainerFactoryFile(file);
```

Reply:
FWIW, I think this is probably OK for a one off, as long as we avoid it in the future.

Reply:
Ah don't think I realized you could access the maven project from a non-pom migration, will update!